File: codes/perl/hello/entry.pl
#! /usr/bin/perl -w

my $n = $ARGV[0];
die "Please enter an integer value" unless (defined $n);

print "Fact($n)= ${\fact($n)}\n";
print "Fact(" . $n . ")= ". fact($n) . "\n";

sub fact {
    my $i = shift;
    return 1 if ($i <= 0);
    return $i * fact($i - 1);
}

File: codes/php/hello/entry.php
#!/usr/bin/php
<?php

if (sizeof($argv) < 2) {
    echo "Please enter an integer value\n";
    exit();
}

$n = $argv[1];
echo "Fact($n)= " . fact($n) . "\n";

function fact($i) {
    if ($i <= 0) return 1;
    return $i * fact($i - 1);
}
?>
